server/server.go

```diff
@@ -20,15 +22,60 @@ import (
 
 	"github.com/spacemeshos/poet/config"
 	"github.com/spacemeshos/poet/gateway"
-	"github.com/spacemeshos/poet/gateway/broadcaster"
 	"github.com/spacemeshos/poet/logging"
 	"github.com/spacemeshos/poet/release/proto/go/rpc/api"
 	"github.com/spacemeshos/poet/rpc"
 	"github.com/spacemeshos/poet/service"
 )
 
+type Server struct {
+	svc          *service.Service
+	cfg          config.Config
+	rpcListener  net.Listener
+	restListener net.Listener
+}
+
+func New(cfg config.Config) (*Server, error) {
+	rpcListener, err := net.Listen(cfg.RPCListener.Network(), cfg.RPCListener.String())
+	if err != nil {
+		return nil, fmt.Errorf("failed to listen: %v", err)
+	}
+
+	restListener, err := net.Listen(cfg.RESTListener.Network(), cfg.RESTListener.String())
```

Comment:
sorry for lack of context. what was the original reason for not using grpc for getting proof, but using restful?

Reply:
Poet provides both "raw" GRPC and REST API via a [grpc-gateway](https://github.com/grpc-ecosystem/grpc-gateway) reverse proxy. The clients are free to choose which one they use.

I don't know why the client in go-spacemesh uses the HTTP REST API 🤷. I created an issue some time ago to change the client to use grpc natively: https://github.com/spacemeshos/go-spacemesh/issues/3702.